errors/errors.go

```diff
@@ -35,10 +35,18 @@ type ValidationError string
 func NewValidationError(str string) error { return ValidationError(str) }
 func (e ValidationError) Error() string   { return string(e) }
 
-// StaleDataError is returned when a rule modification is attempted but the
-// underlying data has changed and the change is no longer valid.
+// StaleDataError is returned when a rule modification can not be completed
+// because rule metadata is no longer valid.
 type StaleDataError string
 
 // NewStaleDataError creates a new version mismatch error.
 func NewStaleDataError(str string) error { return StaleDataError(str) }
 func (e StaleDataError) Error() string   { return string(e) }
+
+// InvalidChangeError is returned when a change is applied to a ruleset,
+// mapping rule, or rollup rule that is invalid.
+type InvalidChangeError string
```

Comment:
`InvalidInputError` might be more generic? If so, perhaps return it where we used to return `RuleConflictError` as well.